OPSEXP-2527: update micrometer doc & dashboards (#2757)
Alfresco/acs-packaging: docs/micrometer/README.md

metrics.restMetricsReporter.enabled=false
metrics.restMetricsReporter.path.enabled=false
```

> This feature is enterprise only!

> We have defined the main property metrics.enabled which defaults to false. If

Reviewer: ```suggestion
We have defined the main property `metrics.enabled` which defaults to `false`. If
```

turned off, the web-script API will return 404.

:warning: In case you want to enable REST metrics with path enabled, make sure
your Prometheus instance can cope with the amount of time series that will be

> created, or ensure relabeling is done in scrapping configuration to reduce the

Reviewer: ```suggestion
created, or ensure relabeling is done in scraping configuration to reduce the
```

> scrapping
I assume this was meant to be `scraping` instead

amount of time series. In the example below, `metricRelabelings` are used to
extract the version, model and endpoint of the REST API from the `servicePath`

> label and then drop the inital label, thus avoiding to create huge amounts of

Reviewer: ```suggestion
label and then drop the inital label, thus avoiding the creation of huge numbers of
```

configured to scrape the endpoint. The following is an example of a Prometheus

> configuration that scrapes the Alfresco endpoint:

Author: This is what is described  line 130 when I describe the `metricRelabelings`. Is it not enough or is it unclear this relate to these regex?

  namespace: acs
spec:
  endpoints:
    - metricRelabelings:
        - action: replace
          regex: >-
            /alfresco/api/-default-/(private|public)/([^/]+)/versions/([0-9]+(\.[0-9]+)*)/([^/]+)(/.*)?
          replacement: $3
          sourceLabels:
            - servicePath
          targetLabel: alfresco_rest_api_version
        - action: replace
          regex: >-
            /alfresco/api/-default-/(private|public)/([^/]+)/versions/([0-9]+(\.[0-9]+)*)/([^/]+)(/.*)?
          replacement: $2
          sourceLabels:
            - servicePath
          targetLabel: alfresco_rest_api_model
        - action: replace
          regex: >-
            /alfresco/api/-default-/(private|public)/([^/]+)/versions/([0-9]+(\.[0-9]+)*)/([^/]+)(/.*)?
          replacement: $5
          sourceLabels:
            - servicePath
          targetLabel: alfresco_rest_api_endpoint
        - action: labeldrop
          regex: servicePath
      path: /alfresco/s/prometheus
      port: http
  jobLabel: alfresco-repo-micrometer
  namespaceSelector:
    any: false
  selector:
    matchLabels:
      app.kubernetes.io/component: alfresco-repository
      app.kubernetes.io/instance: acs
```

> > Where the label selector and namespace should be adjusted to your own deployment.

Reviewer 2: ```suggestion
> `metadata.namespace` and `selector.matchLabels` should be adjusted to your own deployment.
```

```

> `metadata.namespace` and `selector.matchLabels` should be adjusted to your own
  deployment.

### Grafana Dashboards

This folder also contains dashboards that you can import in your
[Grafana](https://grafana.com/) app to visualize various metrics exposed by
Alfresco through a Prometheus aggregation server.

Feel free to update or add new Dashboards or queries that others may find useful:

* [Database Metrics](/docs/micrometer/Alfresco_Enterprise_Database_Dashboard-1714473792698.json)
* [Alfresco REST API Metrics](/docs/micrometer/Alfresco_Enterprise_REST_API_Dashboard-1714475951666.json)
* [Alfresco JVM Metrics](/docs/micrometer/Alfresco_Enterprise_JVM_Dashboard-1714472967588.json)
* [Alfresco Servlet Metrics](/docs/micrometer/Alfresco_Enterprise_Servlet_Dashboard-1714473338586.json)

> The JVM metrics dashboard have been largely inspired from the "JVM micrometer"
> dashboard available at https://grafana.com/dashboards/4701. One can choose to

> reuse this orignal dashboard btu be aware that Alfresco micrometer

Reviewer: ```suggestion
reuse this original dashboard but be aware that Alfresco micrometer
```

Typos

> implementation do not populate the `application` label this dashboard expects.

Reviewer: ```suggestion
implementation does not populate the `application` label this dashboard expects.
```
